### PR TITLE
analysis sort and searchresult

### PR DESCRIPTION
**Pull request by jaserud**
Related issue: https://github.com/icgc-argo/rdpc-gateway/issues/58

Changes:
- Analyses gql qeury returns AnalysisSearchResult
- SearchResult has search info, with totalHits, contentCount and hasNextFrom
- Added simple aggregateAnalyses query

### CODE REVIEW

File: src/main/resources/schema.graphql [hunk 1]
```diff
@@ -149,8 +152,42 @@ type SampleMatchedAnalysisPair {
...
+
+type AggregationResult {
+    totalHits: String!
+}
+
 extend type Query {
-    analyses(filter: AnalysisFilter, page: Page): [Analysis]
+    analyses(filter: AnalysisFilter, page: Page, sorts: [AnalysisSort!]): AnalysesSearchResult!
+    aggregateAnalyses(filter: AnalysisFilter): AggregationResult!
     files(filter: FileFilter, page: Page): [File]
```
Comment (lepsalex) [inline src/main/resources/schema.graphql:191]: I know we don't currently use the file search much but how much work would it be to roll out the same changes for this as well?

**Review by lepsalex — APPROVED** [review]
Overall looks great, one comment about potentially expanding the scope of this to include file search as well.

File: src/main/resources/schema.graphql (hunk 1 above)
Comment (jaserud) [reply src/main/resources/schema.graphql:191]: haha actually yea I'm doing that now, was trying to get it in before you guys had a chance to review :stuck_out_tongue: .. The only thing I'm still trying to figure out is what the default sort should be for files...

Comment (jaserud) [reply src/main/resources/schema.graphql:191]: I'll open in a new PR.. when its ready now :+1:

**Pull request merged**